chain/chunks/src/logic.rs

```diff
@@ -158,47 +158,37 @@ pub fn decode_encoded_chunk(
     shard_tracker: &ShardTracker,
 ) -> Result<(ShardChunk, PartialEncodedChunk), Error> {
     let chunk_hash = encoded_chunk.chunk_hash();
-    let _span = debug_span!(
+    let span = debug_span!(
         target: "chunks",
         "decode_encoded_chunk",
         height_included = encoded_chunk.cloned_header().height_included(),
         shard_id = ?encoded_chunk.cloned_header().shard_id(),
         ?chunk_hash)
     .entered();
 
-    if let Ok(shard_chunk) = encoded_chunk
-        .decode_chunk(epoch_manager.num_data_parts())
-        .map_err(|err| Error::from(err))
-        .and_then(|shard_chunk| {
-            if !validate_chunk_proofs(&shard_chunk, epoch_manager)? {
-                return Err(Error::InvalidChunk);
-            }
-            Ok(shard_chunk)
-        })
-    {
-        debug!(
-            target: "chunks",
-            ?chunk_hash,
-            encoded_length = encoded_chunk.encoded_length(),
-            num_tx = shard_chunk.to_transactions().len(),
-            ?me,
-            "Reconstructed and decoded");
-        let partial_chunk = create_partial_chunk(
-            encoded_chunk,
-            merkle_paths,
-            shard_chunk.prev_outgoing_receipts().to_vec(),
-            me,
-            epoch_manager,
-            shard_tracker,
-        )
-        .map_err(|err| Error::ChainError(err.into()))?;
-
-        Ok((shard_chunk, partial_chunk))
-    } else {
-        // Can't decode chunk or has invalid proofs, ignore it
-        error!(target: "chunks", ?chunk_hash, ?me, "Reconstructed, but failed to decoded chunk");
-        Err(Error::InvalidChunk)
+    let shard_chunk = encoded_chunk.decode_chunk().map_err(|err| {
+        error!(target: "chunks", ?chunk_hash, ?me, "reconstructed, but failed to decode chunk");
+        err
+    })?;
+    if !validate_chunk_proofs(&shard_chunk, epoch_manager)? {
+        return Err(Error::InvalidChunk);
     }
+
+    span.record("chunk_hash", format!("{:?}", chunk_hash));
+    span.record("encoded_length", encoded_chunk.encoded_length());
+    span.record("num_tx", shard_chunk.to_transactions().len());
+    span.record("me", format!("{:?}", me));
```

Comment:
```suggestion
    span.record("encoded_length", encoded_chunk.encoded_length());
    span.record("num_tx", shard_chunk.to_transactions().len());
```

`chunk_hash` is already recorded at the beginning of the span and `me` can be recorded there too.